thavalon-webapp/src/components/gameComponents/lobby.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useEffect, useState } from "react";
+import { Container, ListGroup, Button } from "react-bootstrap";
+import { GameSocket, InboundMessage, InboundMessageType, OutboundMessageType } from "../../utils/GameSocket";
+
+/**
+ * Interface for the Lobby props object.
+ */
+interface LobbyProps {
+    friendCode: string
+}
+
+/**
+ * Component listing players currently in the lobby and a button to start the game.
+ */
+export function Lobby(props: LobbyProps): JSX.Element {
+    // State for maintaining a connection.
+    const [connection, setConnection] = useState<GameSocket | undefined>(undefined);
+    // State for maintaining the player list.
+    const [playerList, setPlayerList] = useState<string[]>([]);
+
+    /**
+     * Handles any lobby messages that come from the server. If the message type
+     * is a PlayerList change, the playerList is updated accordingly.
+     * @param message An incoming message from the server
+     */
+    function handleLobbyMessage(message: InboundMessage): void {
```

Comment:
I'm not 100% sure, but does this need to be inside of the useEffect or outside the component?

Reply:
It works here. I don't think the location of the function matters, as it's still passed in to the subscribe.